src/python.rs

```diff
@@ -99,26 +99,42 @@ impl PartialOrd<(u8, u8, u8)> for PythonVersionInfo<'_> {
     }
 }
 
-/// Marker type that indicates that the GIL is currently held.
+/// A marker token that represents holding the GIL.
```

Comment:
Is it worth expanding on GIL for this introduction? Possibly also with a link to some Python docs.

```suggestion
/// A marker token that represents holding the Python interpreter's Global Interpreter Lock (GIL).
```